TestNewAppSourceAuthRequired

@stevekuznetsov @bparees @csrwng All my recent test and merge runs have failed on 

```
--- FAIL: TestNewAppSourceAuthRequired (0.01s)
15:03:17 	newapp_test.go:1943: 
15:03:17 		*** Please tell me who you are.
15:03:17 		
15:03:17 		Run
15:03:17 		
15:03:17 		  git config --global user.email "you@example.com"
15:03:17 		  git config --global user.name "Your Name"
15:03:17 		
15:03:17 		to set your account's default identity.
15:03:17 		Omit --global to set the identity only in this repository.
15:03:17 		
15:03:17 		fatal: unable to auto-detect email address (got 'root@1dcf384af09c.(none)')
15:03:17 FAIL
```

see https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/20/

--- a/test/integration/newapp_test.go
+++ b/test/integration/newapp_test.go
@@ -1929,7 +1929,12 @@ func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) 
 	}
 
 	// Set initial repo contents
-	gitRepo := git.NewRepository()
+	gitRepo := git.NewRepositoryWithEnv([]string{
+		"GIT_AUTHOR_NAME=developer",
+		"GIT_AUTHOR_EMAIL=developer@example.com",
+		"GIT_COMMITTER_NAME=developer",
+		"GIT_COMMITTER_EMAIL=developer@example.com",
+	})
 	if err = gitRepo.Init(initialRepoDir, false); err != nil {
 		t.Fatalf("%v", err)
 	}

